extensions/sherlock-utils/src/derivable-cache.ts

```diff
@@ -0,0 +1,58 @@
+import { _internal, Derivable, isSettableDerivable, lens, LensDescriptor, SettableDerivable, Unwrappable } from '@politie/sherlock';
+
+export interface MapImplementation<K, V> {
+    set(key: K, value: V): void;
+    delete(key: K): void;
+    get(key: K): V | undefined;
+}
+
+export interface DerivableCacheOptions<K, V> {
+    derivableFactory(key: K): Derivable<V>;
+    mapFactory?(): MapImplementation<K, Derivable<V>>;
+    delayedEviction?: boolean;
+}
+
+export type DerivableCache<K, V> = (key: Unwrappable<K>) => SettableDerivable<V>;
+
+function defaultMapFactory<K, V>(): MapImplementation<K, V> { return new Map; }
+
+export function derivableCache<K, V>(opts: DerivableCacheOptions<K, V>): DerivableCache<K, V> {
+    const cache = (opts.mapFactory || defaultMapFactory)();
+
+    const { delayedEviction, derivableFactory } = opts;
+    const descriptor: LensDescriptor<V, K> = {
+        get: key => {
+            let derivable = cache.get(key);
+            // If the cache has a hit for the current key, we know it is already connected (through another proxy).
+            if (derivable) {
+                return derivable.get();
+            }
+
+            // A cache miss means no other proxy is currently connected.
+            derivable = derivableFactory(key);
+            if (derivable instanceof _internal.Constant) {
+                // Enable connection administration (constants cannot be connected)
+                derivable = derivable.map(v => v);
+            }
+            if (delayedEviction) {
+                derivable.autoCache();
+            }
+            // Get the state of our derivable early so it connects when needed.
+            const state = derivable.getState();
```

Comment:
If a `Derivable` was already in the cache, `.get()` will be called on it, which means that `Derivable` will handle 'unresolved' and 'error' values.

When it is created here, `.getState()` is called, this way it won't throw, but any handling of 'unresolved' or 'error' is done in the `Lens`.

I don't think anyone will notice the difference, but maybe `get` both the same way? For symmetry?

Reply:
If we use `#get()` there, the code after it will not be called in case of unresolved.